mbed_devices/_internal/htm_file.py

```diff
@@ -0,0 +1,63 @@
+"""Facilitates parsing MBED.HTM files found on the Mbed devices.
+
+For legacy reasons, we support many flavours of MBED.HTM files.
+"""
+import pathlib
+import re
+from typing import Type, TypeVar, Optional, List, cast
+
+
+T = TypeVar("T", bound="HTMFileContentsParser")
+
+
+class HTMFileContentsParser:
+    """Extracts information from MBED.HTM file contents."""
+
+    def __init__(self, file_contents: str) -> None:
+        """Initialises HTMFileParser with file contents."""
+        self._file_contents = file_contents
+
+    @property
+    def target_id(self) -> Optional[str]:
+        """Returns target parsed from the file contents, None if not found."""
+        regex = r"""
+            (?:code|auth)=            # attribute name
+            (?P<target>[a-fA-F0-9]+)  # target identifier
+        """
+        match = _search([regex], self._file_contents, re.VERBOSE)
+        if match:
+            return cast(str, match["target"])
+        return None
+
+    @property
+    def daplink_version(self) -> Optional[str]:
+        """Returns version parsed from the file contents, None if not found."""
+        as_semver = r"<!--.*Version: (?P<version>\d+\.\d+\.\d+)\.?.*-->"
+        as_digits = r"<!--.*Version: (?P<version>\d+).*-->"
+        match = _search([as_semver, as_digits], self._file_contents)
+        if match:
+            return cast(str, match["version"])
+        return None
+
+    @property
+    def daplink_build(self) -> Optional[str]:
+        """Returns build num/date parsed from file contents, None if not found."""
+        as_digits = r"<!--.*build (?P<build>\d+).*-->"
+        as_date = r"<!--.*Build: (?P<build>.* \d{2}:\d{2}:\d{2}).*-->"
+        match = _search([as_digits, as_date], self._file_contents)
+        if match:
+            return cast(str, match["build"])
+        return None
+
+    @classmethod
+    def build_for_path(cls: Type[T], file_path: str) -> T:
```

Comment:
Should we just call this classmethod `from_file`? I think the current name is a bit confusing.